protocol/xmrtaker/message_handler.go

```diff
@@ -148,17 +150,26 @@ func (s *swapState) handleSendKeysMessage(msg *net.SendKeysMessage) (net.Message
 	// start goroutine to check that XMRMaker locks before t_0
 	go func() {
 		// TODO: this variable is so that we definitely refund before t0.
-		// this will vary based on environment (eg. development should be very small,
-		// a network with slower block times should be longer)
-		const timeoutBuffer = time.Second * 5
-		until := time.Until(s.t0)
-
-		log.Debugf("time until refund: %vs", until.Seconds())
+		// Current algorithm is to trigger the timeout when only 15% of the allotted
+		// time is remaining. If the block interval is 1 second on a test network and
+		// and T0 is 7 seconds after swap creation, we need the refund to trigger more
+		// than one second before the block with a timestamp exactly equal to T0 to
+		// satisfy the strictly less than requirement. 7s * 15% = 1.05s. 15% remaining
+		// may be reasonable even with large timeouts on production networks, but more
+		// research is needed.
+		t0Delta := s.t1.Sub(s.t0) // time between swap start and T0 is equal to T1-T0
```

Comment:
it doesn't make a difference now since the time between initiation and t0 is the same as between t0 and t1, but this should probably be the time difference between t0 and initiation

Reply:
I would have done that, but I didn't have the information to compute it. We'll need to store the timestamp of the block that created the swap in `swapState` to compute it that way.